File: gamelogic/src/main/java/jashburn00/tastelesswizardry/classes/Freeze.java
public class Freeze extends Spell {
    public Freeze(){
        super("Freeze", 0, 45);
    }

    @Override
    public void useOn(Hero h, Entity e){
        if(h.mana >= this.manaCost){
            h.mana -= this.manaCost;
            e.isFrozen = true;
            e.frozenTurns = 2;
            System.out.println(h.name+" froze "+e.name+"! (costed "+this.manaCost+" mana)");
        } else {
            System.out.println(h.name+" tried to use "+this.name+", but had insufficient mana!");
        }
    }
}

File: scripts/Freeze.js
import {Spell} from './Spell.js';

export class Freeze extends Spell {
    constructor(){
        super("Freeze", 0, 45, "Freezes the enemy for two turns.");
    }

    useOn(h, e){
        if(h.mana >= this.manaCost){
            h.mana -= this.manaCost;
            e.isFrozen = true;
            e.frozenTurns = 2;
            return h.name+" froze "+e.name+"! (costed "+this.manaCost+" mana)";
        } else {
            return h.name+" tried to use "+this.name+", but had insufficient mana!";
        }
    }
}
